File: experiments/llm-reasoning-cost/generated_code/java/05_file_io_mini.java
```java
import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;

public class FileIO {

    public static Result writeGreeting(String filename) {
        BufferedWriter writer = null;
        try {
            writer = new BufferedWriter(new FileWriter(filename));
            writer.write("Hello from the program!");
            return Result.success();
        } catch (IOException e) {
            return Result.error(e.getMessage());
        } finally {
            if (writer != null) {
                try {
                    writer.close();
                } catch (IOException e) {
                    return Result.error(e.getMessage());
                }
            }
        }
    }

    public static class Result {
        private final boolean success;
        private final String errorMessage;

        private Result(boolean success, String errorMessage) {
            this.success = success;
            this.errorMessage = errorMessage;
        }

        public static Result success() {
            return new Result(true, null);
        }

        public static Result error(String errorMessage) {
            return new Result(false, errorMessage);
        }

        public boolean isSuccess() {
            return success;
        }

        public String getErrorMessage() {
            return errorMessage;
        }
    }
}
```

File: experiments/llm-reasoning-cost/generated_code/c/05_file_io_mini.c
#include <stdio.h>
#include <stdlib.h>

typedef enum {
    SUCCESS,
    FILE_ERROR
} Result;

Result write_greeting(const char *filename) {
    FILE *file = fopen(filename, "w");
    if (file == NULL) {
        return FILE_ERROR;
    }

    if (fputs("Hello from the program!\n", file) == EOF) {
        fclose(file);
        return FILE_ERROR;
    }

    if (fclose(file) != 0) {
        return FILE_ERROR;
    }

    return SUCCESS;
}
